astropy/units/quantity_helper.py

```diff
@@ -448,18 +449,29 @@ def helper_degree_minute_second_to_radian(f, unit1, unit2, unit3):
 # https://docs.scipy.org/doc/scipy/reference/special.html
 
 try:
+    import scipy
     import scipy.special as sps
 except ImportError:
-    pass
+    HAS_SCIPY = False
 else:
-    # ufuncs that require dimensionless input and give dimensionless output
-    dimensionless_to_dimensionless_sps_ufuncs = (
-        sps.erf, sps.gamma, sps.loggamma, sps.gammasgn,
+    HAS_SCIPY = True
+    dimensionless_to_dimensionless_sps_ufuncs = [
+        sps.erf, sps.gamma, sps.gammasgn,
         sps.psi, sps.rgamma, sps.erfc, sps.erfcx, sps.erfi, sps.wofz,
         sps.dawsn, sps.entr, sps.exprel, sps.expm1, sps.log1p, sps.exp2,
         sps.exp10, sps.j0, sps.j1, sps.y0, sps.y1, sps.i0, sps.i0e, sps.i1,
         sps.i1e, sps.k0, sps.k0e, sps.k1, sps.k1e, sps.itj0y0,
-        sps.it2j0y0, sps.iti0k0, sps.it2i0k0)
+        sps.it2j0y0, sps.iti0k0, sps.it2i0k0]
+
+# TODO: Revert https://github.com/astropy/astropy/pull/7219 when astropy
+#       requires scipy>=0.18.
+# See https://github.com/astropy/astropy/issues/7159
+HAS_SCIPY_18 = HAS_SCIPY and minversion(scipy, "0.18")
```

Comment:
I wouldn't have define `HAS_SCIPY`, but rather put this definition at the bottom of the try. Then everything else can go under the else clause just as it was before rather than  have the extra ifs.
But this is nitpicking, so I'm OK with whatever @saimn prefers.

Reply:
I copied the incantation from `astropy/modeling/tests/test_models.py` but I can always change it back. Let's see what @saimn says. Thanks! :smile:

Reply:
yeah, but here we don't need to use `HAS_SCIPY` anywhere ;)

Reply:
Yes, good point. I'll go ahead and make the change as requested then.

Reply:
Agreed with @bsipocz, so the current version looks good.